eventsourcing/src/test/java/org/axonframework/eventsourcing/configuration/AnnotatedEventSourcedEntityBuilderTest.java

```diff
@@ -30,79 +30,87 @@
 
 /**
- * Test class validating the {@link AnnotatedEventSourcedEntityBuilder}.
+ * Test class validating the {@link AnnotatedEventSourcedEntityModule}.
  *
  * @author Mitchell Herrijgers
  * @author Steven van Beelen
  */
 class AnnotatedEventSourcedEntityBuilderTest {
 
     private Configuration parentConfiguration;
+    private StubLifecycleRegistry lifecycleRegistry;
 
     @BeforeEach
     void setUp() {
         parentConfiguration = EventSourcingConfigurer.create().build();
+        lifecycleRegistry = new StubLifecycleRegistry();
     }
 
     @Test
     void annotatedEntityThrowsNullPointerExceptionForNullIdentifierType() {
         //noinspection DataFlowIssue
-        assertThrows(NullPointerException.class, () -> EventSourcedEntityBuilder.annotatedEntity(null, Course.class));
+        assertThrows(NullPointerException.class, () -> EventSourcedEntityModule.annotated(null, Course.class));
     }
 
     @Test
     void annotatedEntityThrowsNullPointerExceptionForNullEntityType() {
         //noinspection DataFlowIssue
-        assertThrows(NullPointerException.class, () -> EventSourcedEntityBuilder.annotatedEntity(CourseId.class, null));
+        assertThrows(NullPointerException.class, () -> EventSourcedEntityModule.annotated(CourseId.class, null));
     }
 
     @Test
     void annotatedEntityThrowsIllegalArgumentExceptionForNotAnnotatedEntity() {
         assertThrows(IllegalArgumentException.class,
-                     () -> EventSourcedEntityBuilder.annotatedEntity(CourseId.class, CourseId.class));
+                     () -> EventSourcedEntityModule.annotated(CourseId.class, CourseId.class));
     }
 
     @Test
     void entityNameCombinesIdentifierAndEntityTypeNames() {
         String expectedEntityName = "Course#CourseId";
 
-        EventSourcedEntityBuilder<CourseId, Course> testSubject =
-                EventSourcedEntityBuilder.annotatedEntity(CourseId.class, Course.class);
+        EventSourcedEntityModule<CourseId, Course> testSubject =
+                EventSourcedEntityModule.annotated(CourseId.class, Course.class);
 
         assertEquals(expectedEntityName, testSubject.entityName());
     }
 
     @Test
     void repositoryConstructsEventSourcingRepositoryForEntityFactory() {
-        EventSourcedEntityBuilder<CourseId, Course> testSubject =
-                EventSourcedEntityBuilder.annotatedEntity(CourseId.class, Course.class);
+        EventSourcedEntityModule.annotated(CourseId.class, Course.class)
+                                .build(parentConfiguration, lifecycleRegistry);
+        lifecycleRegistry.start(parentConfiguration);
 
-        Repository<CourseId, Course> result = testSubject.repository()
-                                                         .build(parentConfiguration);
+        StateManager stateManager = parentConfiguration.getComponent(StateManager.class);
+        Repository<CourseId, Course> result = stateManager.repository(Course.class, CourseId.class);
```

Comment:
You can also directly retrieve the `Repository` **if** you know the entity name, right?

Reply:
So, you mean by retrieving the module from the `AxonConfiguration`, and then `.getComponent(Repository.class)`? Yes!

Reply:
To be a bit more exact, from the `Module`, not the parent `AxonConfiguration`.